### PR TITLE
feat: TreeSelect support `showSearch` in multiple mode

### LINKED ISSUE #15921: TreeSelect multiple mode should also support `showSearch={false}`
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Same as single mode.

### What does the proposed API look like?
<TreeSelect showSearch={false} {...props} />

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

close #15921 

### 💡 Solution

Update rc-tree-select to support that.

### 📝 Changelog

TreeSelect support `showSearch` in multiple mode.

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/tree-select/index.zh-CN.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ title: TreeSelect
 | searchValue | 搜索框的值，可以通过 `onSearch` 获取用户输入 | string | - |
 | treeIcon | 是否展示 TreeNode title 前的图标，没有默认样式，如设置为 true，需要自行定义图标相关样式 | boolean | false |
 | showCheckedStrategy | 定义选中项回填的方式。`TreeSelect.SHOW_ALL`: 显示所有选中节点(包括父节点). `TreeSelect.SHOW_PARENT`: 只显示父节点(当父节点下所有子节点都选中时). 默认只显示子节点. | enum{TreeSelect.SHOW_ALL, TreeSelect.SHOW_PARENT, TreeSelect.SHOW_CHILD } | TreeSelect.SHOW_CHILD |
-| showSearch | 在下拉中显示搜索框(仅在单选模式下生效) | boolean | false |
+| showSearch | 是否显示搜索框 | boolean | false |
```
Comment (afc163) [inline components/tree-select/index.zh-CN.md]: 是否支持搜索

File: components/tree-select/index.en-US.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ Any data whose entries are defined in a hierarchical manner is fit to use this c
 | searchValue | work with `onSearch` to make search value controlled. | string | - |
 | treeIcon | Shows the icon before a TreeNode's title. There is no default style; you must set a custom style for it if set to `true` | boolean | false |
 | showCheckedStrategy | The way show selected item in box. **Default:** just show child nodes. **`TreeSelect.SHOW_ALL`:** show all checked treeNodes (include parent treeNode). **`TreeSelect.SHOW_PARENT`:** show checked treeNodes (just show parent treeNode). | enum { TreeSelect.SHOW_ALL, TreeSelect.SHOW_PARENT, TreeSelect.SHOW_CHILD } | TreeSelect.SHOW_CHILD |
-| showSearch | Whether to display a search input in the dropdown menu(valid only in the single mode) | boolean | false |
+| showSearch | Whether to display a search input | boolean | false |
```
Comment (afc163) [inline components/tree-select/index.en-US.md]: 还有这里

Comment (zombieJ) [reply components/tree-select/index.en-US.md]: 突然感觉还是应该叫做“是否显示搜索框” ，这个 prop 叫做 `showSearch` 。
支持搜索 / support search 应该是 `searchable`。

Comment (afc163) [reply components/tree-select/index.en-US.md]: 支持搜索在目前的场景里更准确一点，虽然属性名是 `showSearch`。

**Comment by zombieJ** [comment]
antd 中默认 `showSearch` 为 `false`。支持后，TreeSelect 默认是没有 Search 的。这点感觉要调整一下。文档里描述也要改成：
```
`false` when single | `true` when multiple
```

File: components/tree-select/__tests__/__snapshots__/demo.test.js.snap [hunk 3]
```diff
@@ -96,27 +96,6 @@ exports[`renders ./components/tree-select/demo/checkable.md correctly 1`] = `
...
-            type="text"
-            value=""
-          />
-          <span
-            class="ant-select-search__field__mirror"
-          >
-             
-          </span>
-        </span>
-      </li>
```
Comment (afc163) [inline components/tree-select/__tests__/__snapshots__/demo.test.js.snap]: 这个咋变了？

**Comment by zombieJ** [comment]
发错到 master 了，revert 掉。

**Pull request merged**